pontoon/base/models.py

```diff
@@ -2330,6 +2358,58 @@ def cleaned_key(self):
     def __str__(self):
         return self.string
 
+    def stats_states_map(self, locale):
+        translations = list(
+            self.translation_set
+                .filter(locale=locale)
+                .prefetch_related('errors', 'warnings')
+        )
+
+        approved_strings_count = len([
+            t for t in translations
+            if t.approved
+        ])
+        fuzzy_strings_count = len([
+            t for t in translations
+            if t.fuzzy
+        ])
+        unrevieved_count = len([
+            t for t in translations
+            if not (t.approved or t.fuzzy or t.rejected)
+        ])
+        errors_count = len([
+            t for t in translations
+            if (t.approved or t.fuzzy) and t.errors.exists()
+        ])
+        warnings_count = len([
+            t for t in translations
+            if (t.approved or t.fuzzy) and t.warnings.exists()
+        ])
+
+        if self.string_plural:
+            approved_state = (approved_strings_count == locale.nplurals)
+            fuzzy_state = (fuzzy_strings_count == locale.nplurals)
+        else:
+            approved_state = (approved_strings_count == 1)
+            fuzzy_state = (fuzzy_strings_count == 1)
+
+        if not (approved_state or fuzzy_state):
+            errors_state = (errors_count > 0)
+            warnings_state = (warnings_count > 0)
+        else:
+            errors_state = False
+            warnings_state = False
```

Comment:
These values should be `0` instead of `False`, right?

Reply:
That's a good question, currently this function returns booleans and later they're casted into integers.
I can do this more explicit :-)